src/main/java/seedu/address/model/itinerary/ItineraryAttraction.java

```diff
@@ -93,9 +93,8 @@ public boolean isSameItineraryAttraction(ItineraryAttraction otherItineraryAttra
      * Returns false if timing does not clash with this itinerary attraction.
      */
     public boolean isTimingClash(ItineraryAttraction itineraryAttraction) {
-        // Might make a mistake here with the ! due to De Morgan's law
-        return !this.getStartTime().isEarlierThan(itineraryAttraction.getStartTime())
-                && !this.getEndTime().isLaterThan(itineraryAttraction.getEndTime());
+        return (this.getStartTime().isEarlierThan(itineraryAttraction.getEndTime()))
+                && (itineraryAttraction.getStartTime().isEarlierThan(this.getEndTime()));
```

Comment:
With `&&,` you only return `true` if `this attraction` start/end time is between the start/end time of `itineraryAttraction`? Should it be `||`